lib/Dialect/FIRRTL/NLATable.cpp

```diff
@@ -147,3 +147,7 @@ void NLATable::eraseModule(StringAttr name) {
   symToOp.erase(name);
   nodeMap.erase(name);
 }
+
+void NLATable::addModule(Operation *op) {
+  symToOp[static_cast<FModuleLike>(op).moduleNameAttr()] = op;
```

Comment:
nit: Can we use llvm `cast`? Also, if `op` is always FModuleLike in every call site, can we just change the argument type to FModuleLike?

Reply:
Completely agree, updated it to FModuleLike.